executor/runcexecutor/executor_linux.go

```diff
@@ -0,0 +1,308 @@
+package runcexecutor
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"sync"
+	"time"
+
+	"github.com/containerd/console"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	runcV2 "github.com/containerd/containerd/runtime/v2/runc"
+	"github.com/containerd/containerd/sys/reaper"
+	runc "github.com/containerd/go-runc"
+	"github.com/moby/buildkit/executor"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sys/unix"
+)
+
+var setupReaper sync.Once
+
+func handleSignals(ctx context.Context) error {
+	signals := make(chan os.Signal, 32)
+	signal.Notify(signals, unix.SIGPIPE, unix.SIGCHLD)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
Do we need to `signal.Stop(signals)` and `close(signals)` here?

Reply:
yeah, will add.  If the runc worker is enabled this loop will effectively run forever though.